### PR TITLE
Module 1 - React Components

### PR DESCRIPTION
**Pull request by YauheniAntsipenka**
image
image

DEPLOYED APP
https://yauheniantsipenka.github.io/react_course/

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
Comment (Margaryta-Maletz) [inline public/index.html:5]: Add favicon.ico in the public folder.

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (Margaryta-Maletz) [inline public/index.html]: Delete this link, because this picture was deleted

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
```
Comment (Margaryta-Maletz) [inline public/index.html]: The same as above. Delete this link, because this json file was deleted.

File: src/common/Button/button.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    font-weight: 700;
+    font-size: 16px;
+    line-height: 34px;
+    
+    background: #007298;
+    border-radius: 5px;
+    border: 0px;
+    font-size: 16px;
+    color: #fff;
+}
```
Comment (Margaryta-Maletz) [inline src/common/Button/button.scss]: Add last empty line, please, here and in all html, ts, tsx, scss files

File: src/common/Button/button.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    font-family: 'Open Sans';
+    font-style: normal;
+    font-weight: 700;
+    font-size: 16px;
+    line-height: 34px;
+    
+    background: #007298;
+    border-radius: 5px;
+    border: 0px;
+    font-size: 16px;
```
Comment (Margaryta-Maletz) [inline src/common/Button/button.scss]: Property font-size duplicates property font-size

File: src/common/Button/button.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    font-family: 'Open Sans';
+    font-style: normal;
+    font-weight: 700;
+    font-size: 16px;
+    line-height: 34px;
+    
+    background: #007298;
+    border-radius: 5px;
+    border: 0px;
```
Comment (Margaryta-Maletz) [inline src/common/Button/button.scss]: Unit of measure 'px' is redundant
```suggestion
    border: 0;
```

File: src/common/Button/button.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.button {
+    width: 150px;
+    height: 40px;
+    margin-right: 2%;
+
+    font-family: 'Open Sans';
```
Comment (Margaryta-Maletz) [inline src/common/Button/button.scss]:
```suggestion
    font-family: 'Open Sans', serif;
```

File: src/components/Courses/components/CourseCard/coursecard.scss [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: flex-start;
+    align-content: center;
+
+    background: #FFFFFF;
+    
+    border: 1px solid rgba(0, 0, 0, 0.12);
+    box-shadow: 10px 0px 0px -5px rgba(0, 0, 0, 0.8) inset, 4px 4px 3px rgba(0, 0, 0, 0.2), 0px -8px 5px -5px rgba(0, 0, 0, 0.2);
```
Comment (Margaryta-Maletz) [inline src/components/Courses/components/CourseCard/coursecard.scss]: Unit of measure 'px' is redundant
```suggestion
    box-shadow: 10px 0 0 -5px rgba(0, 0, 0, 0.8) inset, 4px 4px 3px rgba(0, 0, 0, 0.2), 0 -8px 5px -5px rgba(0, 0, 0, 0.2);
```

File: src/App.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import { mockedCoursesList } from './constants';
+
+import { Courses } from './components/Courses/Courses';
+import { CourseInfo } from './components/CourseInfo/CourseInfo';
+import { Header } from './components/Header/Header';
+import { CourseCardProps } from './components/Courses/components/CourseCard/CourseCard.types';
+import { getDuration } from './helpers/getCourseDuration';
+import { getAuthors } from './helpers/getAuthorsNames';
+
+import './app.scss';
```
Comment (Margaryta-Maletz) [inline src/App.tsx:12]:
```suggestion
import './App.scss';
```

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { Logo } from './components/Logo/Logo';
+import { Button } from '../../common/Button/Button';
+
+import { HeaderProps } from './Header.types';
+
+import './header.scss';
```
Comment (Margaryta-Maletz) [inline src/components/Header/Header.tsx:8]:
```suggestion
import './Header.scss';
```

File: src/components/Header/components/Logo/Logo.tsx [hunk 11]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+import mainLogo from '../../../../resources/logo.png';
+
+import './logo.scss';
```
Comment (Margaryta-Maletz) [inline src/components/Header/components/Logo/Logo.tsx:5]:
```suggestion
import './Logo.scss';
```

File: src/components/CourseInfo/CourseInfo.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+					<div className='courseInfoGeneralTitle'>
+						<h2>{courseToShow.title}</h2>
+					</div>
+					<div className='courseInfoNote'>
+						<div className='courseInfoCard'>
+							<div className='courseInfoDescription'>
+								<div className='courseInfoDescriptionElements'>
+									<h3>
+										Description: <br />
+									</h3>
```
Comment (Margaryta-Maletz) [inline src/components/CourseInfo/CourseInfo.tsx]: <br /> is redundant
```suggestion
									<h3>Description:</h3>
```

File: src/components/CourseInfo/CourseInfo.tsx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+									<span>{courseToShow.description}</span>
```
Comment (Margaryta-Maletz) [inline src/components/CourseInfo/CourseInfo.tsx]:
```suggestion
									<p>{courseToShow.description}</p>
```

File: src/components/CourseInfo/CourseInfo.tsx [hunk 14]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import { Button } from '../../common/Button/Button';
+
+import { CourseInfoProps } from './CourseInfo.types';
+
+import './courseinfo.scss';
+
+export const CourseInfo: React.FC<CourseInfoProps> = (props) => {
```
Comment (Margaryta-Maletz) [inline src/components/CourseInfo/CourseInfo.tsx:9]: Oh, this component has too much levels.
Try my idea and let's discuss it.
```
export const CourseInfo: React.FC<CourseInfoProps> = (props) => {
	const courseToShow = props.courseCard;

	const ADDITIONAL_INFO = [
		{
			id: 'ID:',
			value: courseToShow.id,
		},
		{
			id: 'Duration:',
			value: courseToShow.duration,
		},
		{
			id: 'Created:',
			value: courseToShow.creationDate.toLocaleDateString('ru-RU'),
		},
		{
			id: 'Authors:',
			value: courseToShow.authors,
		},
	];

	return (
		<div className='courseInfoBody'>
			<div className='courseInfoGeneral'>
				<div className='courseInfoGeneralTitle'>
					<h2>{courseToShow.title}</h2>
				</div>
				<div className='courseInfoNote'>
					<div className='courseInfoCard'>
						<h3>Description:</h3>
						<p>{courseToShow.description}</p>
					</div>
					<div className='courseInfoAdditionalInfoGeneralGroup'>
						{ADDITIONAL_INFO.map((item) => {
							return (
								<div key={item.id} className={'courseInfoAdditionalInfoGroup'}>
									<div className={'courseInfoAdditionalInfoCategoryName'}>
										{item.id}
									</div>
									<div>{item.value}</div>
								</div>
							);
						})}
					</div>
				</div>

				<div className='courseInfoAdditionalInfoButtonsGroup'>
					<Button
						text='BACK'
						onClickFunction={() => {
							props.changeState('start', '');
						}}
					/>
				</div>
			</div>
		</div>
	);
};
```

File: src/components/CourseInfo/courseinfo.scss [hunk 15]
```diff
@@ -0,0 +1,142 @@
+.courseInfoBody {
```
Comment (Margaryta-Maletz) [inline src/components/CourseInfo/courseinfo.scss:1]: Of course, for my solution need different scss
```
.courseInfoBody {
    width: 100%;
    height: 50%;

    display: inline-flex;
    justify-content: center;
}

.courseInfoGeneral {
    width: 80%;

    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin-top: 2%;
}

.courseInfoGeneralTitle {
    display: flex;
    align-items: flex-start;
}

.courseInfoNote {
    display: flex;
    justify-content: center;

    background: #FFFFFF;
    
    border: 1px solid rgba(0, 0, 0, 0.12);
    box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.2) inset;
    border-radius: 5px;
}

.courseInfoCard {
    width: 60%;
    margin-left: 5%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;

    margin-right: 1%;
    border-right: 1px solid #CFCFCF;
    padding: 1%;

    color: #333E48;
}

.courseInfoAdditionalInfoGeneralGroup {
    width: 40%;
    margin-left: 2%;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    gap: 2%;
}

.courseInfoAdditionalInfoGroup {
    width: 90%;

    display: flex;
    gap: 2%;
    justify-content: flex-start;
}

.courseInfoAdditionalInfoCategoryName {
    min-width: 100px;
    font-weight: bold;
    color: #333E48;
}

.courseInfoAdditionalInfoButtonsGroup {
    margin-top: 2%;
    margin-left: auto;

    display: inline-flex;
    align-items: flex-end;
}
```

File: src/components/Courses/components/CourseCard/CourseCard.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (Margaryta-Maletz) [inline src/components/Courses/components/CourseCard/CourseCard.tsx:1]: It will be better to change this component and delete redundant div and span

**Review by Margaryta-Maletz — APPROVED** [review]
Good start!
In my opinion, we have to discuss some simple FE rules.
Generally, I agree with your self-estimation 93%

File: public/index.html (hunk 1 above)
Comment (YauheniAntsipenka) [reply public/index.html:5]: thanks, done

File: public/index.html (hunk 2 above)
Comment (YauheniAntsipenka) [reply public/index.html]: deleted

File: public/index.html (hunk 3 above)
Comment (YauheniAntsipenka) [reply public/index.html]: deleted

File: src/common/Button/button.scss (hunk 4 above)
Comment (YauheniAntsipenka) [reply src/common/Button/button.scss]: refactored

File: src/common/Button/button.scss (hunk 5 above)
Comment (YauheniAntsipenka) [reply src/common/Button/button.scss]: removed

File: src/common/Button/button.scss (hunk 6 above)
Comment (YauheniAntsipenka) [reply src/common/Button/button.scss]: refactored

File: src/common/Button/button.scss (hunk 7 above)
Comment (YauheniAntsipenka) [reply src/common/Button/button.scss]: thanks, refactored

File: src/components/Courses/components/CourseCard/coursecard.scss (hunk 8 above)
Comment (YauheniAntsipenka) [reply src/components/Courses/components/CourseCard/coursecard.scss]: thanks, refactored

File: src/App.tsx (hunk 9 above)
Comment (YauheniAntsipenka) [reply src/App.tsx:12]: I will rename scss file. thanks

File: src/components/Header/Header.tsx (hunk 10 above)
Comment (YauheniAntsipenka) [reply src/components/Header/Header.tsx:8]: the same here

File: src/components/Header/components/Logo/Logo.tsx (hunk 11 above)
Comment (YauheniAntsipenka) [reply src/components/Header/components/Logo/Logo.tsx:5]: and the same here :)

File: src/components/CourseInfo/CourseInfo.tsx (hunk 12 above)
Comment (YauheniAntsipenka) [reply src/components/CourseInfo/CourseInfo.tsx]: refactored

File: src/components/CourseInfo/CourseInfo.tsx (hunk 13 above)
Comment (YauheniAntsipenka) [reply src/components/CourseInfo/CourseInfo.tsx]: thanks, refactored

**Pull request merged**